price-feeder/oracle/provider/binance.go

```diff
@@ -100,35 +103,48 @@ func NewBinanceProvider(
 		Path:   binanceWSPath,
 	}
 
-	wsConn, resp, err := websocket.DefaultDialer.Dial(wsURL.String(), nil)
-	defer resp.Body.Close()
-	if err != nil {
-		return nil, fmt.Errorf(
-			types.ErrWebsocketDial.Error(),
-			ProviderBinance,
-			err,
-		)
-	}
+	binanceLogger := logger.With().Str("provider", string(ProviderBinance)).Logger()
 
 	provider := &BinanceProvider{
-		wsURL:           wsURL,
-		wsClient:        wsConn,
-		logger:          logger.With().Str("provider", string(ProviderBinance)).Logger(),
+		logger:          binanceLogger,
 		endpoints:       endpoints,
 		tickers:         map[string]BinanceTicker{},
 		candles:         map[string][]BinanceCandle{},
 		subscribedPairs: map[string]types.CurrencyPair{},
 	}
 
-	if err := provider.SubscribeCurrencyPairs(pairs...); err != nil {
-		return nil, err
-	}
+	provider.setSubscribedPairs(pairs...)
 
-	go provider.handleWebSocketMsgs(ctx)
+	controller := NewWebsocketController(
+		ctx,
+		ProviderBinance,
+		wsURL,
+		provider.getSubscriptionMsgs(),
+		provider.messageReceived,
+		time.Duration(0),
```

Comment:
Let's use a constant called something like `disabledPing` instead of `time.Duration(0)`, it's a little bit more obvious what you're doing

Reply:
agreed. https://github.com/umee-network/umee/pull/1554/commits/ef919f977998d74fd192688a008d607876dec610